### PR TITLE
Convert Expressions/Functions to Holonomic Functions.

### PR DESCRIPTION
**Pull request by shubhamtibra**
- [x] Convert Meijer-G functions to Holonomic.
- [x] Convert Expressions/Functions to Holonomic.
- [x] Tests

### CODE REVIEW

**Comment by shubhamtibra** [comment]
Ping @jksuom @certik .

File: sympy/holonomic/holonomic.py [hunk 1]
```diff
@@ -1085,6 +1098,79 @@ def _find_conditions(simp, x, x0, order, evalf=False):
     return HolonomicFunction(sol, x).composition(z)
 
 
+def from_meijerg(func, x0=0, evalf=False):
+    """
+    Converts a Meijer-G function to Holonomic.
```
Comment (jksuom) [inline sympy/holonomic/holonomic.py]: Meijer G-function

**Comment by shubhamtibra** [comment]
The method `from_sympy` works for cases where `meijerint._rewrite1` can convert the expression/function to `meijerg`. Let me know about your views. Ping @jksuom @certik .

**Comment by jksuom** [comment]
I think that it would suffice. The rewrite table in `meijerint` can probably be extended, but that could be another project.

**Comment by certik** [comment]
This looks great to me. Very useful.

**Comment by shubhamtibra** [comment]
The checks failed giving a strange error `Fatal Python error: Cannot recover from stack overflow.` in `sympy/core/tests/test_wester.py`. [Here](https://travis-ci.org/sympy/sympy/jobs/135052948).

**Comment by jksuom** [comment]
It is probably not related to this PR. I've restarted the job.

**Comment by shubhamtibra** [comment]
Thanks @jksuom .

**Comment by shubhamtibra** [comment]
I think this can be merged. Also, what would be the next thing to do/implement?

**Comment by jksuom** [comment]
I think so too. @certik ?

**Comment by certik** [comment]
Remove the `[WIP]` when you think it's ready to merge.

**Comment by shubhamtibra** [comment]
Thanks @certik . I forgot to remove the `[WIP]`. You have any thoughts for what should be done next?

**Comment by certik** [comment]
What to do next: How about the numerical evaluation as we talked about? Unless you have something else that is of higher priority?

**Comment by shubhamtibra** [comment]
I think we can go with numerical evaluation. What in it: An another numerical method (like RK4) or the upper layer for generating the set of points.

**Comment by certik** [comment]
Ah, I missed that you already did that! Euler is enough for now, but I can see just a simple test there:
```python
def test_evalf():
    x = symbols('x')
    R, Dx = DifferentialOperators(QQ.old_poly_ring(x), 'Dx')
    # a straight line on real axis
    r = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1]
    p = HolonomicFunction((1 + x)*Dx**2 + Dx, x, 0, [0, 1])
    s = '0.699525841805253'
    assert sstr(p.evalf(r)[-1]) == s
    # a traingle with vertices (0, 1+i, 2)
    r = [0.1 + 0.1*I]
    for i in range(9):
        r.append(r[-1]+0.1+0.1*I)
    for i in range(10):
        r.append(r[-1]+0.1-0.1*I)
    s = '1.07530466271334 - 0.0251200594793912*I'
    assert sstr(p.evalf(r)[-1]) == s
    p = HolonomicFunction(Dx**2 + 1, x, 0, [0, 1])
    s = '0.905546532085401 - 6.93889390390723e-18*I'
    assert sstr(p.evalf(r)[-1]) == s
    # a rectangular path (0 -> i -> 2+i -> 2)
    r = [0.1*I]
    for i in range(9):
        r.append(r[-1]+0.1*I)
    for i in range(20):
        r.append(r[-1]+0.1)
    for i in range(10):
        r.append(r[-1]-0.1*I)
    p = HolonomicFunction(Dx**2 + 1, x, 0, [1,1]).evalf(r)
    s = '0.501421652861245 - 3.88578058618805e-16*I'
    assert sstr(p[-1]) == s
```
I would test it on some functions, say `sin(x)`, `cos(x)`, and make sure that as you refine the mesh (points), that you get a correct value, say for x=pi/2 or something, given initial conditions at x=0.

Then we should add a test that the answer does not depend on the path, unless you go around a pole.

I am now playing with your module, I'll suggest more things to test. If you get bored, yes, try to add the rk4 method, that's a standard method, which should be accurate enough for our purposes.

I'll ping you with more stuff to do.

**Comment by shubhamtibra** [comment]
I will add these tests. I am also looking for bugs and will be working on them. Thanks.

**Pull request merged**